### ISSUE TITLE
bot.registry not always defined after creating bot with createBot

### ISSUE THREAD

**Issue by IceTank** [PrismarineJS/mineflayer#2875]
- [x] The [FAQ](https://github.com/PrismarineJS/mineflayer/blob/master/docs/FAQ.md) doesn't contain a resolution to my issue 

## Versions
 - mineflayer: latest

## Detailed description of a problem
mineflayer-pathfinder relies on the fact that bot.registry exists when loading the plugin. mineflayer-pathfinder is suppose to be able to be loaded before the bot has spawned with the spawn event. But in some cases bot.registry is not defined after createBot was called. I am not 100% how this could be fixed but the bot.registry definition should at least be changed to reflect the fact that bot.registry can be undefined after bot creation.

## Expected behavior
index.d.ts to reflect the fact that bot.registry can be undefined

**Comment by IceTank** [comment]
I think this issue can also happen with other functions on bot. One possible solution I see is an event that emits a bot object when all properties of bot are defined. This would also be useful for full type safety. Basically make all default properties on the bot returned by createBot possibly undefined and all properties on bot in this new event strictly defined.

**Comment by extremeheat** [comment]
If there's an undefined error that means something is not waiting for the correct event (such as login or spawn). See the examples.

**Comment by IceTank** [comment]
The use case I have is this:
```js
const mf = require('mineflayer')
const bot = mf.createBot()

console.info(bot.registry) // undefined
```
This is a pitfall I think a lot off people fall for. And not unique to registry either. Most other defined properties on bot will also be undefined.

**Comment by rom1504** [comment]
> mineflayer-pathfinder relies on the fact that bot.registry exists when loading the plugin

this should be fixed in mineflayer-pathfinder then

**Comment by Zn10plays** [comment]
> This is a pitfall I think a lot off people fall for. And not unique to registry either. Most other defined properties on bot will also be undefined.

Yeah, this is very common, especially with the bot methods, and people new to MF usually have problems with this. Not having a proper error also does not help either. I feel like there should at least be a layer in between the method calls and the bot that checks if all the requirements for the actions are met.
